src/com/google/javascript/jscomp/ConvertChunksToESModules.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2011 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.javascript.jscomp;
+
+import com.google.javascript.jscomp.NodeTraversal.AbstractPostOrderCallback;
+import com.google.javascript.jscomp.parsing.parser.FeatureSet;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+import com.google.javascript.rhino.Token;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Finds all references to global symbols in a different output chunk and add ES Module imports and
+ * exports for them.
+ *
+ * <p>Given the name of the global object is NS
+ *
+ * <pre> var a = 1; function b() { return a }</pre>
+ *
+ * becomes
+ *
+ * <pre> var a = 1; var b = function b() { return a }; export {a};</pre>
```

Comment:
Fly on the wall comment: why did `b` switch to a `var` and why isn't it exported?

Reply:
I'll address - it's a copy / paste error from the RescopeGlobalSymbols pass. It wasn't exported because it was not referenced in a different chunk.